src/main/kotlin/org/eclipse/uprotocol/uri/validator/UriValidator.kt

```diff
@@ -28,140 +28,170 @@ import org.eclipse.uprotocol.v1.UAuthority
 import org.eclipse.uprotocol.v1.UResource
 import org.eclipse.uprotocol.v1.UUri
 import org.eclipse.uprotocol.validation.ValidationResult
-import java.util.Objects
 
 /**
  * class for validating Uris.
  */
-interface UriValidator {
-    companion object {
-        /**
-         * Validate a [UUri] to ensure that it has at least a name for the uEntity.
-         *
-         * @param uri [UUri] to validate.
-         * @return Returns UStatus containing a success or a failure with the error message.
-         */
-        fun validate(uri: UUri): ValidationResult {
-            if (isEmpty(uri)) {
-                return ValidationResult.failure("Uri is empty.")
-            }
-            if (uri.hasAuthority() && !isRemote(uri.authority)) {
-                return ValidationResult.failure("Uri is remote missing uAuthority.")
-            }
-            return if (uri.entity.name.isBlank()) {
-                ValidationResult.failure("Uri is missing uSoftware Entity name.")
-            } else ValidationResult.success()
+object UriValidator {
+    /**
+     * Validate a [UUri] to ensure that it has at least a name for the uEntity.
+     *
+     * @param uri [UUri] to validate.
+     * @return Returns UStatus containing a success or a failure with the error message.
+     */
+    fun validate(uri: UUri): ValidationResult {
```

Comment:
This could be provided as an extension function on UUri (same comment for all the methods in this UriValidator

Reply:
Addressed in [PR20](https://github.com/eclipse-uprotocol/up-kotlin/pull/20)